osu.Server.Queues.ScoreStatisticsProcessor/Processors/MedalAwarders/ModIntroductionMedalAwarder.cs

```diff
@@ -0,0 +1,69 @@
+﻿using JetBrains.Annotations;
+using MySqlConnector;
+using osu.Game.Online.API.Requests.Responses;
+using osu.Game.Rulesets;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Osu.Mods;
+using osu.Server.Queues.ScoreStatisticsProcessor.Helpers;
+using osu.Server.Queues.ScoreStatisticsProcessor.Models;
+using System.Collections.Generic;
+
+namespace osu.Server.Queues.ScoreStatisticsProcessor.Processors.MedalAwarders
+{
+    [UsedImplicitly]
+    public class ModIntroductionMedalAwarder : IMedalAwarder
+    {
+        public bool RunOnFailedScores => false;
+
+        public IEnumerable<Medal> Check(SoloScoreInfo score, UserStats userStats, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            // Ensure the mod is the only one selected
+            if (score.Mods.Length != 1)
+                yield break;
+
+            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(score.RulesetID);
+            Mod mod = score.Mods[0].ToMod(ruleset);
+
+            // Ensure the mod is in the default configuration
+            if (!mod.UsesDefaultConfiguration)
+                yield break;
+
+            foreach (var medal in medals)
+            {
+                if (checkMedal(score, medal, mod))
+                    yield return medal;
+            }
+        }
+
+        private bool checkMedal(SoloScoreInfo score, Medal medal, Mod mod)
+        {
+            return medal.achievement_id switch
```

Comment:
This switchexpr usage is most definitely not required. I'm fine with it personally but others may not be.

Reply:
While it's not required for the existing medals per se, I would argue it's pretty nice to have regardless, considering mods can now have customized settings, so the conditions here could get more complex (for example a 2x dt medal, or special DA medals etc.)

Reply:
Let's keep this for now.